### PR TITLE
Overhaul AppData handling between DEBUG and RELEASE mode and fix #479

### LINKED ISSUE #479: Appdata crash when running a new version without running the executable first
When you run a new compiled version without running the executable from the installer first, it crashes since it tries to access the appdata directory which doesn't exist

### PR DESCRIPTION
**Pull request by GustaveHerve**
Allows to build in true DEBUG mode by default and RELEASE mode using the appropriate arguments of dev.py (./dev.py build -b r). We now have truly distinct AppData for both modes (local AppData in DEBUG mode, true user AppData in RELEASE mode).
convolution_kernels, input_filters, and shaders directories were also moved to AppData for consistency.

If we are on RELEASE mode, the program will copy default files of its "reference" folders to the real AppData which will allow them to be used by the user.

This implies that the program is still doing some of the AppData handling at runtime which may not be ideal, but circumventing this is not that easy (we would need to have the installer propagate the files in the real AppData which the CMake build system doesn't allow us to do directly)

There are two duplicates of the get_exe_dir() function used, this should also be fixed (we could have global variable storing the result of get_exe_dir() and accessible everywhere to avoid redudants calls).

### CODE REVIEW

File: Holovibes/sources/gui/windows/MainWindow.cc [hunk 1]
```diff
@@ -118,8 +118,7 @@ MainWindow::MainWindow(QWidget* parent)
     // TODO: move in AppData
```
Comment (alexispinson) [inline Holovibes/sources/gui/windows/MainWindow.cc:118]: Maybe you can remove the todo here.

File: Holovibes/sources/main.cc [hunk 2]
```diff
@@ -123,63 +123,20 @@ static void print_help(holovibes::OptionsParser parser)
 // Copy all files from src path to dest path (the directories will be created if not exist)
 static void copy_files(const std::filesystem::path src, std::filesystem::path dest)
 {
-    if (std::filesystem::exists(dest))
-        return;
+    // if (std::filesystem::exists(dest))
```
Comment (alexispinson) [inline Holovibes/sources/main.cc]: Delete comments here also maybe

**Review by alexispinson — CHANGES_REQUESTED** [review]
Good code just also they may be conflict with simon branch for reducing leaks

**Review by Baschtien — APPROVED** [review]

File: Holovibes/includes/holovibes_config.hh [hunk 3]
```diff
@@ -5,6 +5,30 @@
 #pragma once
 
 #include "camera_config.hh"
+#include <Windows.h>
+#include <filesystem>
+
+// FIXME: get rid of this duplicate get_exe_dir, find a way to import the original one in tools.hh or move it to a
+// a place from which we can import it easily
+static std::string get_exe_dir()
+{
```
Comment (Hpn4) [inline Holovibes/includes/holovibes_config.hh:14]: C'est exactement la même fonction que dans camera_utils.cc peut être merge les deux dans un seul fichier pour eviter les doublons.

**Review by Hpn4 — COMMENTED** [review]
Il y a deux fois exactement le même get_exe_dir. Peut etre le rajouter à un fichier dans tools ou common. Je trouve que les noms des maccros rendent le code plus compiqué à lire. Peut être simplifier le noms des maccros.

File: Holovibes/sources/gui/windows/MainWindow.cc [hunk 4]
```diff
@@ -532,10 +523,10 @@ void MainWindow::set_preset_file_on_gpu()
 {
     std::filesystem::path dest = __PRESET_FOLDER_PATH__ / "FILE_ON_GPU.json";
     // Check if we are in DEBUG or RELEASE
-    if (!std::filesystem::exists(dest))
-    {
-        dest = std::filesystem::path(get_exe_dir()).parent_path().parent_path() / "Preset" / "FILE_ON_GPU.json";
-    }
+    // if (!std::filesystem::exists(dest))
```
Comment (TitouanGragnic) [inline Holovibes/sources/gui/windows/MainWindow.cc:526]: deadcode ?

Comment (TitouanGragnic) [reply Holovibes/sources/gui/windows/MainWindow.cc:526]: fix

**Pull request merged**